app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt

```diff
@@ -1214,6 +1214,160 @@ class PinPasswordActivityTest {
     }
   }
 
+  @Test
+  fun testPinPassword_withAdmin_inputPin_configChange_inputPinIsPersisted() {
+    ActivityScenario.launch<PinPasswordActivity>(
+      PinPasswordActivity.createPinPasswordActivityIntent(
+        context = context,
+        adminPin = adminPin,
+        profileId = adminId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .perform(editTextInputAction.appendText("1234"), closeSoftKeyboard())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .check(matches(withText("1234")))
+    }
+  }
+
+  @Test
+  fun testPinPassword_withUser_inputPin_configChange_inputPinIsPersisted() {
+    ActivityScenario.launch<PinPasswordActivity>(
+      PinPasswordActivity.createPinPasswordActivityIntent(
+        context = context,
+        adminPin = adminPin,
+        profileId = userId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .perform(editTextInputAction.appendText("123"), closeSoftKeyboard())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .check(matches(withText("123")))
+    }
+  }
+
+  @Test
+  fun testPinPassword_withAdmin_inputPin_exceedMaxLength_textIsTrimmed() {
```

Comment:
```suggestion
  fun testPinPassword_adminUser_inputPinExceedsFive_textIsTrimmedToFive() {
```

This sets clearer expectaions of the test setup and outcome

Reply:
Updated.